packages/react-testing/src/root.tsx

```diff
@@ -274,21 +298,21 @@ export class Root<Props> implements Node<Props> {
     }
 
     this.ensureRoot();
-    this.act(() => this.reactRoot!.unmount());
+    return this.act(() => this.reactRoot!.unmount());
   }
 
   async destroy() {
     const {element, mounted} = this;
+    let mountedPromise;
 
     if (mounted) {
-      this.unmount();
+      mountedPromise = this.unmount();
     }
     element.remove();
     connected.delete(this);
     this.destroyed = true;
-    this.actPromiseResolver();
-    // flush the micro task to wait until react commits all pending updates.
-    await this.actPromise;
+    await mountedPromise;
+    this.actCallbacks.forEach((callback) => callback());
```

Comment:
Won't these need to be awaited as well? And shouldn't they be invoked prior to unmounting?

Reply:
> Won't these need to be awaited as well?

These callbacks will `popActScope` and `recursivelyFlushAsyncActWork` if needed. There's nothing to await there.

> And shouldn't they be invoked prior to unmounting?

Once the root is destroyed there should be no more assertions on it, we're calling these to flush the queue out and it's not really related to this instance of the wrapper itself. I tried moving them up but it fails the tests as it appears that the final act in the `unmount` needs to be called and awaited on first.